controllers/tempo/sa_filter.go

```diff
@@ -0,0 +1,52 @@
+package controllers
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/grafana/tempo-operator/internal/manifests/manifestutils"
+)
+
+// Filter service account objects already created and modified by OCP. e.g  bound service account tokens
+// when generating pull secrets adds an annotation to the SA. In such case we are not interested on modified it.
+func filterServiceAccountObjects(ctx context.Context,
```

Comment:
Would it make sense to add any kind of test for this?

Reply:
Sure!

Reply:
Due the latest modifications. I don't think we require any other test. Do we?